tests/integration/index/query_with_unique_index_on_relation_filter_test.go

```diff
@@ -70,3 +70,106 @@ func TestQueryWithUniqueCompositeIndex_WithFilterOnIndexedRelation_ShouldFilter(
 
 	testUtils.ExecuteTestCase(t, test)
 }
+
+func TestQueryWithUniqueCompositeIndex_WithIndexComprising2RelationsAndFilterOnIt_ShouldFilter(t *testing.T) {
```

Comment:
suggestion: It might be highlighting with a comment why this test is important to have, for example compared to just for composite scalar tests.

Please also document the significance of including two id fields, and one object field in the index.  Perhaps highlighting why a test with two object fields in the index (e.g. `includes: [{name: "owner"}, {name: "manufacturer"}` is not required.

Reply:
The test case is actually a bit simpler. I wanted to update it, but forgot. Thanks for reminding.

I will add some comments.

Reply:
Thanks Islam, looks good!